js/modules/k6/execution/execution.go

```diff
@@ -297,72 +307,81 @@ func optionsAsObject(rt *goja.Runtime, options lib.Options) (*goja.Object, error
 }
 
 type tagsDynamicObject struct {
-	Runtime *goja.Runtime
-	State   *lib.State
+	runtime *goja.Runtime
+	state   *lib.State
+}
+
+func (o *tagsDynamicObject) isIndexed(key string) goja.Value {
+	tcv := o.state.Tags.GetCurrentValues()
+	if _, ok := tcv.Tags.Get(key); ok {
+		return o.runtime.ToValue(true)
+	}
+	if _, ok := tcv.Metadata[key]; ok {
+		return o.runtime.ToValue(false)
+	}
+	return goja.Undefined()
```

Comment:
Ahhh the famous three valued boolean :tada: 


Not certain this won't be more confusing then not - I did expect that this will tell me if `url` is indexed but it gave me `undefined` which was super confusing for me. I guess with some documentation it will be easier. 

And I guess we should have another API that answeres teh question "is this tag globally indexable or not".

Reply:
Yeah... This API only "knows" about the VU-level tags, which aren't all that many. I even hesitated adding the API, since pretty much all such non-indexed tags will be set by the users, so they probably don't really need a way to query them...

> And I guess we should have another API that answeres teh question "is this tag globally indexable or not".

That may be a... complicated question to answer :sweat_smile: Theoretically, someone could disable an indexed system tag with `--system-tags` and then manually supply tag values with the same name... They will probably shoot themselves in the foot doing that, but they could, and I have no idea what such an API should return then... :sweat_smile:

Reply:
Any extra comments here or can this be resolved?